src/internal/storage/chunk/doc.go

```diff
@@ -0,0 +1,18 @@
+/*
+Package chunk is Pachyderm's interface with object storage.
+
+A chunk is the basic unit of storage that PFS uses for arbitrary data.
+Chunks are identified by the hash of the underlying data, which, along with content-defined chunking, is the mechanism of PFS deduplication.
+Content-defined chunking is implemented by [chunk.ComputeChunks].
+This package also supports a second way of chunking data, [chunk.Batcher], used for small files where chunk overhead outweighs deduplication benefits.
+
+# Backwards compatibility
+
+Some implementation details might not make sense based on the current code (e.g. the [chunk.Ref.Edge] check in [chunk.StableDataRefs]).
+The algorithms for file and index chunking have changed in 2.x, and we must support previously-written data.
+Here are some examples of conditions in past data which current code will not generate:
+  - a file may be split across multiple chunks, some of which also contain other files
```

Comment:
To be clear, a file can still be split across multiple chunks, but those chunks never have other file content when that is the case.

Reply:
ah right, I initially had this list in the opposite direction (things that aren't true) and didn't think about this being misleading when flipped